WcaOnRails/app/controllers/users_controller.rb

```diff
@@ -13,11 +13,16 @@ def index
     respond_to do |format|
       format.html { }
       format.json do
-        @users = User
-        if params[:search]
-          @users = @users.where("name LIKE :input OR wca_id LIKE :input OR email LIKE :input", { input: "%#{params[:search]}%" })
+        @users = User.joins("INNER JOIN Countries ON iso2 = country_iso2")
+        params[:search]&.split&.each do |part|
+          like_query = %w(users.name wca_id email Countries.name).map do |column|
+            column + " LIKE :part"
+          end.join(" OR ")
+          @users = @users.where(like_query, { part: "%#{part}%" })
         end
-        if params[:sort]
+        if params[:sort] == "country"
+          @users = @users.order("Countries.name #{params[:order] == "asc" ? "asc" : "desc"}")
```

Comment:
Oh wow, I didn't think for a second about injection possibilities. Yeah, posting `"ASC; DELETE FROM RESULTS WHERE wca_id= 2005FLEI01"` into the naive implementation of this would probably be bad.

Some thoughts:

- If Active Record simply allowed `.order("Countries.name" => params[:order])` there wouldn't be a problem.
- Fixing this one place is good, but doesn't secure the whole app. Maybe it's worth using Brakeman: https://github.com/presidentbeef/brakeman
- There is no test for this, so someone clueless like me could easily make the code unsafe with a simple refactoring.